packages/legend-extension-dsl-service/src/stores/studio/UpdateProjectServiceQuerySetupStore.ts

```diff
@@ -167,8 +180,69 @@ export class UpdateProjectServiceQuerySetupStore {
     }
   }
 
+  setIsCurrentProjectNotConfigured(val: boolean | undefined): void {
```

Comment:
as said before, I think we could abstract this logic into a sub-state. See my sugestion for `ProjectConfigurationStatusCheckState`